solidity/security/missing-array-lengths-check.yaml

```diff
@@ -0,0 +1,55 @@
+rules:
+  - id: missing-array-lengths-check
+    message: User activities may not be completed successfully if the arrays' lengths aren't determined to be the same.
+    metadata:
+      category: security
+      tags:
+        - array
+        - length
+    patterns:
+        - pattern: |
+              function $F(..., $TYPE[] $VAR, ..., $TYPE2[] $VAR2, ...) {
```

Comment:
This pattern is very vague, $VAR and $VAR2 may not be related to each other. This rule tries to find an absence of some conditions which is not robust since you have to describe all situations.
With that said, I am not sure how to approach this vulnerability in some other way using semgrep 🤔